utilities/backupable/backupable_db_test.cc

```diff
@@ -1303,7 +1303,7 @@ TEST_F(BackupableDBTest, TableFileWithDbChecksumCorruptedDuringBackup) {
     OpenDBAndBackupEngine(true /* destroy_old_data */, false /* dummy */, sopt);
 
     FillDB(db_.get(), 0, keys_iteration);
-    bool corrupted = false;
+
```

Comment:
Will make `corrupted` an atomic variable work?

Reply:
Thanks for the suggestion! I tried using an atomic variable but it seems it doesn't work here.

Reply:
qq: will `corrupted` always be true afterwards? In which case it can remain `false`?

I tried with/without the following. Running the test 100 times seems to show that the issue can be fixed.
```
diff --git a/utilities/backupable/backupable_db_test.cc b/utilities/backupable/backupable_db_test.cc
index 81e4fd456..77addbd55 100644
--- a/utilities/backupable/backupable_db_test.cc
+++ b/utilities/backupable/backupable_db_test.cc
@@ -1303,7 +1303,7 @@ TEST_F(BackupableDBTest, TableFileWithDbChecksumCorruptedDuringBackup) {
     OpenDBAndBackupEngine(true /* destroy_old_data */, false /* dummy */, sopt);

     FillDB(db_.get(), 0, keys_iteration);
-    bool corrupted = false;
+    std::atomic<bool> corrupted{false};
     // corrupt files when copying to the backup directory
     SyncPoint::GetInstance()->SetCallBack(
         "BackupEngineImpl::CopyOrCreateFile:CorruptionDuringBackup",
@@ -1312,13 +1312,13 @@ TEST_F(BackupableDBTest, TableFileWithDbChecksumCorruptedDuringBackup) {
             Slice* d = reinterpret_cast<Slice*>(data);
             if (!d->empty()) {
               d->remove_suffix(1);
-              corrupted = true;
+              corrupted.store(true, std::memory_order_relaxed);
             }
           }
         });
     SyncPoint::GetInstance()->EnableProcessing();
     Status s = backup_engine_->CreateNewBackup(db_.get());
-    if (corrupted) {
+    if (corrupted.load(std::memory_order_relaxed)) {
       ASSERT_NOK(s);
     } else {
       // should not in this path in normal cases
```

Therefore, depending on

Reply:
Thanks! I realized I made a mistake when I modified the code this morning... :
```
    if (corrupted.load(std::memory_order_relaxed)) {
      ASSERT_NOK(backup_engine_->CreateNewBackup(db_.get()));
    } else {
      ASSERT_OK(backup_engine_->CreateNewBackup(db_.get()));
    }
```
which makes the atomic variable pointless >_<

`corrupted` is always true except when there are no table files in the backup to be created or all the table files are empty.

Reply:
SG.